fs/nova/balloc.c

```diff
@@ -746,8 +752,12 @@ static long nova_alloc_blocks_in_free_list(struct super_block *sb,
 
 	/* Try huge block allocation for data blocks first */
 	if (IS_DATABLOCKS_2MB_ALIGNED(num_blocks, atype)) {
-		found_hugeblock = nova_alloc_superpage(sb, free_list,
+		ret_blocks = nova_alloc_superpage(sb, free_list,
 					num_blocks, new_blocknr, from_tail);
+		if (ret_blocks > 0 && new_blocknr != 0) {
```

Comment:
new_blocknr is a pointer. You want to check *new_blocknr?

Reply:
Yes, I'll change it to *new_blocknr != 0.